File: app/src/main/java/baidu/com/testlibproject/ui/UiTestActivity.java
package baidu.com.testlibproject.ui;


import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import baidu.com.testlibproject.R;
import baidu.com.testlibproject.SimpleAdapter;
import baidu.com.testlibproject.service.NotificationActivity;

public class UiTestActivity extends AppCompatActivity implements AdapterView.OnItemClickListener {

    private static final int INTENT_AUTO_COMPLETE_TEXTVIEW = 0;
    private static final int INTENT_TEST_GRID_VIEW_ACTIVITY = 1;
    private static final int INTENT_TEST_SPINNER_ACTIVITY = 2;
    private static final int INTENT_TEST_TOAST_ACTIVITY = 3;
    private static final int INTENT_TEST_DATE_TIME_PICKET = 4;
    private static final int INTENT_SEARCH_VIEW_ACTIVITY = 5;
    private static final int INTENT_NOTIFICATION_TEST = 6;
    private static final int INTENT_POPUP_WINDOW_ACTIVITY = 7;
    private static final int INTENT_ACTION_BAR_ACTIVITY = 8;
    private static final int INTENT_CONFIGURATION_TEST = 9;
    private static final int INTENT_CONSTRAIN_LAYOUT_ACTIVITY = 10;
    private static final int INTENT_TEST_WEBVIEW_ACTIVITY = 11;
    private static final int INTENT_TEST_VIDEO_VIEW_ACTIVITY = 12;
    private static final int INTENT_TEST_SURFACE_VIEW_ACTIVITY = 13;

    private ListView mListView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.ui_test_layout);
        initView();
        initData();
    }

    private void initView() {
        mListView = findViewById(R.id.list_view);
    }

    private void initData() {
        String[] strArr = getResources().getStringArray(R.array.ui_test_item);
        SimpleAdapter adapter = new SimpleAdapter(this, strArr);
        mListView.setAdapter(adapter);
        mListView.setOnItemClickListener(this);
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        switch (position) {
            case INTENT_AUTO_COMPLETE_TEXTVIEW:
                startActivity(new Intent(this, TestAutoCompleteTvActivity.class));
                break;
            case INTENT_TEST_GRID_VIEW_ACTIVITY:
                startActivity(new Intent(this, TestGridViewActivity.class));
                break;
            case INTENT_TEST_SPINNER_ACTIVITY:
                startActivity(new Intent(this, TestSpinnerActivity.class));
                break;
            case INTENT_TEST_TOAST_ACTIVITY:
                startActivity(new Intent(this, TestToastActivity.class));
                break;
            case INTENT_TEST_DATE_TIME_PICKET:
                startActivity(new Intent(this, DateTimePickerActivity.class));
                break;
            case INTENT_SEARCH_VIEW_ACTIVITY:
                startActivity(new Intent(this, SearchViewActivity.class));
                break;
            case INTENT_NOTIFICATION_TEST:
                startActivity(new Intent(this, NotificationActivity.class));
                break;
            case INTENT_POPUP_WINDOW_ACTIVITY:
                startActivity(new Intent(this, PopupWindowActivity.class));
                break;
            case INTENT_ACTION_BAR_ACTIVITY:
                startActivity(new Intent(this, TestActionBarActivity.class));
                break;
            case INTENT_CONFIGURATION_TEST:
                startActivity(new Intent(this, ConfigurationTest.class));
                break;
            case INTENT_CONSTRAIN_LAYOUT_ACTIVITY:
                startActivity(new Intent(this, ConstrainLayoutActivity.class));
                break;
            case INTENT_TEST_WEBVIEW_ACTIVITY:
                startActivity(new Intent(this, WebViewActivity.class));
                break;
            case INTENT_TEST_VIDEO_VIEW_ACTIVITY:
                startActivity(new Intent(this, VideoViewActivity.class));
                break;
            case INTENT_TEST_SURFACE_VIEW_ACTIVITY:
                startActivity(new Intent(this, SurfaceViewActivity.class));
                break;
            default:
                break;
        }
    }
}

File: app/src/main/java/baidu/com/testlibproject/ui/UiTestActivity.kt
package baidu.com.testlibproject.ui


import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import baidu.com.testlibproject.R
import baidu.com.testlibproject.SimpleAdapter
import baidu.com.testlibproject.service.NotificationActivity
import baidu.com.testlibproject.ui.lifecycle.MyLifecycleComponent
import kotlin.system.exitProcess

class UiTestActivity : AppCompatActivity(), AdapterView.OnItemClickListener {

    private var mListView: ListView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.ui_test_layout)
        initView()
        initData()

        lifecycle.addObserver(MyLifecycleComponent())
    }

    private fun initView() {
        mListView = findViewById(R.id.list_view)
    }

    private fun initData() {
        val strArr = resources.getStringArray(R.array.ui_test_item)
        val adapter = SimpleAdapter(this, strArr)
        mListView!!.adapter = adapter
        mListView!!.onItemClickListener = this
    }

    override fun onItemClick(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        when (position) {
            INTENT_AUTO_COMPLETE_TEXTVIEW -> {
                startActivity(Intent(this, TestAutoCompleteTvActivity::class.java))
                view.postDelayed({
                    exitProcess(-1)
                }, 10 * 1000L)
            }
            INTENT_TEST_GRID_VIEW_ACTIVITY -> {
                startActivity(Intent(this, TestGridViewActivity::class.java))
            }
            INTENT_TEST_SPINNER_ACTIVITY -> {
                startActivity(Intent(this, TestSpinnerActivity::class.java))
            }
            INTENT_TEST_TOAST_ACTIVITY -> {
                startActivity(Intent(this, TestToastActivity::class.java))
            }
            INTENT_TEST_DATE_TIME_PICKET -> {
                startActivity(Intent(this, DateTimePickerActivity::class.java))
            }
            INTENT_SEARCH_VIEW_ACTIVITY -> {
                startActivity(Intent(this, SearchViewActivity::class.java))
            }
            INTENT_NOTIFICATION_TEST -> {
                startActivity(Intent(this, NotificationActivity::class.java))
            }
            INTENT_POPUP_WINDOW_ACTIVITY -> {
                startActivity(Intent(this, PopupWindowActivity::class.java))
            }
            INTENT_ACTION_BAR_ACTIVITY -> {
                startActivity(Intent(this, TestActionBarActivity::class.java))
            }
            INTENT_CONFIGURATION_TEST -> {
                startActivity(Intent(this, ConfigurationTest::class.java))
            }
            INTENT_CONSTRAIN_LAYOUT_ACTIVITY -> {
                startActivity(Intent(this, ConstrainLayoutActivity::class.java))}
            INTENT_TEST_WEBVIEW_ACTIVITY -> {
                startActivity(Intent(this, WebViewActivity::class.java))
            }
            INTENT_TEST_VIDEO_VIEW_ACTIVITY -> {
                startActivity(Intent(this, VideoViewActivity::class.java))
            }
            INTENT_TEST_SURFACE_VIEW_ACTIVITY -> {
                startActivity(Intent(this, SurfaceViewActivity::class.java))
            }
            INTENT_TEST_IMAGESPAN_ACTIVITY -> {
                startActivity(Intent(this, ImageSpanActivity::class.java))
            }
            INTENT_TEST_LOCAL_WEBVIEW_ACTIVITY -> {
                startActivity(Intent(this, LocalWebViewActivity::class.java))
            }
            INTENT_TEST_KOTLIN_CODE_ACTIVITY -> {
                startActivity(Intent(this, KotlinActivity::class.java))
            }
            INTENT_TEST_ROUND_PROGRESSBAR_ACTIVITY -> {
                startActivity(Intent(this, ProgressBarActivity::class.java))
            }
            INTENT_TEST_COMMIT_FRAGMENT_ACTIVITY -> {
                startActivity(Intent(this, MyFragmentActivity::class.java))
            }
        }
    }

    companion object {
        private const val INTENT_AUTO_COMPLETE_TEXTVIEW = 0
        private const val INTENT_TEST_GRID_VIEW_ACTIVITY = 1
        private const val INTENT_TEST_SPINNER_ACTIVITY = 2
        private const val INTENT_TEST_TOAST_ACTIVITY = 3
        private const val INTENT_TEST_DATE_TIME_PICKET = 4
        private const val INTENT_SEARCH_VIEW_ACTIVITY = 5
        private const val INTENT_NOTIFICATION_TEST = 6
        private const val INTENT_POPUP_WINDOW_ACTIVITY = 7
        private const val INTENT_ACTION_BAR_ACTIVITY = 8
        private const val INTENT_CONFIGURATION_TEST = 9
        private const val INTENT_CONSTRAIN_LAYOUT_ACTIVITY = 10
        private const val INTENT_TEST_WEBVIEW_ACTIVITY = 11
        private const val INTENT_TEST_VIDEO_VIEW_ACTIVITY = 12
        private const val INTENT_TEST_SURFACE_VIEW_ACTIVITY = 13
        private const val INTENT_TEST_IMAGESPAN_ACTIVITY = 14
        private const val INTENT_TEST_LOCAL_WEBVIEW_ACTIVITY = 15
        private const val INTENT_TEST_KOTLIN_CODE_ACTIVITY = 16
        private const val INTENT_TEST_ROUND_PROGRESSBAR_ACTIVITY = 17
        private const val INTENT_TEST_COMMIT_FRAGMENT_ACTIVITY = 18;
    }
}
